packages/x-components/src/x-modules/tagging/wiring.ts

```diff
@@ -145,12 +152,15 @@ export const taggingWiring = createWiring({
     setQueryTaggingDebounce
   },
   SearchTaggingChanged: {
-    trackQueryWire
+    setQueryTaggingInfo
   },
   UserClickedAResult: {
     trackResultClickedWire
   },
   UserClickedResultAddToCart: {
     trackAddToCartWire
+  },
+  SearchTaggingEmitted: {
+    trackQueryWire
```

Comment:
I was really interested about this so I made some research about it:

According to the spec, during the process of navigate to a new resource the active document is aborted ([step 15](https://html.spec.whatwg.org/multipage/browsing-the-web.html#navigate)) and aborting a document should cancel any instances of fetch ([step 2](https://html.spec.whatwg.org/multipage/browsing-the-web.html#abort-a-document)) 

The [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/fetch) provides the option `keepalive` to prevent a request from being cancelled. [Chrome 80](https://developers.google.com/web/updates/2019/12/chrome-80-deps-rems) recommends this approach or using the Beacon API in case you want to make a request when the document is about to be unloaded, since they disabled the option to do a synchronous XHR.

Google analytics docs describes a way to [specify the transport mechanism](https://developers.google.com/analytics/devguides/collection/analyticsjs/sending-hits#specifying_different_transport_mechanisms) where you can set the beacon API. They even mentioned that the other supported methods face the issue: 

> hits are often not sent if the page is being unloaded

I run a local (dumb) experiment to see if it was possible to reach a server with a request juste before the unload of the document, see results:

  | Chrome | Safari | Firefox | Opera | Vivaldi
-- | -- | -- | -- | -- | --
Fetch | ✅ | ❌ | ❌ | ✅ | ✅
Fetch + keep alive | ✅ | ❌ | ❌ | ✅ | ✅
Beacon | ✅ | ✅ | ✅ | ✅ | ✅

Given that the main idea of this callback event is to allow the consumer of x-components to use its own tracking system, I think we are safe here.

Reply:
Thanks once again for this research. I knew about the possibility of being cancelled, but TBH I hadn't read it from the spec 🤣. Knowing that GA allows to use beacon is also a great discovery, so I agree, we should be safe.